### DISCUSSION TITLE
是否可以做到精灵对比以加快推送速度

### DISCUSSION THREAD

**Discussion by MR-XieXuan** [Bodmer/TFT_eSPI#2471] [Q&A - Sprites]
在程序中有一个精灵,对实际屏幕操作只会操作这一个精灵。
其他的精灵都是对这个精灵做修改。
推送这只精灵的时候只推送修改的部分,而不推送没有修改的部分。以提高推送速度。

**Comment by MR-XieXuan** [comment]
There is a Sprite in the program, and only this Sprite will be manipulated for actual screen operations.

All the other sprites make changes to this Sprite.

When pushing this Sprite, only push the modified part, not the unmodified part. To improve push speed.

**Reply by Bodmer** [reply]
I am not sure what the question is.

Maybe this is the function you need:
https://github.com/Bodmer/TFT_eSPI/blob/master/Extensions/Sprite.h#L127

It pushes part of a sprite to the screen.

**Reply by MR-XieXuan** [reply]
我的意思是,有两个精灵。一个精灵只用作保存当前屏幕的状态。程序实际使用的是另一只精灵。
程序修改另一只精灵后需要对另一只精灵进行推送。推送时会与第一只精灵做比对。并且只推送这两个精灵不相同的部分。
推送完成后保证这两只精灵的值与屏幕完全一致。

I mean, there are two elves.    A Sprite(1) is used only to save the state of the current screen.    The program actually uses a different Sprite(2). 
When the application modifies another Sprite(2), it needs to push another Sprite(1).    When pushed, it is compared to the first Sprite(1).    And only push the different parts of the two sprites(1&2). 
Make sure that the values of the sprites(1&2) are exactly the same as those on the screen after pushing.

**Comment by Bodmer** [comment]
OK

What is the question?

**Comment by MR-XieXuan** [comment]
当我把精灵推向精灵时会出现问题。
所以我没有办法利用Spring来做Buffer处理。
@Bodmer 
```c++

#include <TFT_eSPI.h> 
#include <SPI.h>

#define BUG true

TFT_eSPI tft = TFT_eSPI();  
TFT_eSprite img = TFT_eSprite(&tft);
TFT_eSprite img2 = TFT_eSprite(&img);


void setup() {
  tft.init();
  tft.setRotation(0);

  tft.fillScreen(TFT_BLUE);
  img.createSprite(100, 100);
  img.fillSprite(TFT_RED);
  
  #if BUG
  img2.createSprite(50, 50);
  img2.fillSprite(TFT_GREEN);
  img2.pushSprite(0,0);
  #endif
  
  img.pushSprite(0,0);
}

void loop() {
}
```

**Reply by MR-XieXuan** [reply]
> 当我把精灵推向精灵时会出现问题。
> 所以我没有办法利用Spring来做Buffer处理。
English :
There is a problem when I push sprites towards sprites.
So I can't use Spring to do Buffer processing.

**Reply by MR-XieXuan** [reply]
Sorr, I discover pushToSprite ;

**Reply by MR-XieXuan** [reply]
但是我认为如果可以向我的代码上的那种形式去向精灵推送另一个精灵使用起来会更加方便，而且有利于使用了这个库的项目的维护。

**Reply by MR-XieXuan** [reply]
> 但是我认为如果可以向我的代码上的那种形式去向精灵推送另一个精灵使用起来会更加方便，而且有利于使用了这个库的项目的维护。

English :
But I think it would be easier to use and better to maintain projects that use the library if I could push another Sprite to that form on my code

**Reply by Bodmer** [reply]
OK, but I still do not know what the question is!
